### PR TITLE
Extending JSON utils

### PR DESCRIPTION
**Pull request by jb-perrier**
This PR aims to extend the json utils : 
- Free json object
- Convert json object to string
- Create, modify easily json object using helper funcs

The needs come from the implementation of LSP #567  that require extensive manipulation of JSON objects.

* Edit : In order to keep code size small as possible only necessary func will be added ( json_free ), json should be built directly as string ( hello scratch_buffer ).

### CODE REVIEW

**Comment by jb-perrier** [comment]
In order to keep code size small as possible only necessary func will be added ( json_free ), json should be built directly as string ( hello scratch_buffer ).

**Comment by data-man** [comment]
I think [json.h](https://github.com/sheredom/json.h) is the best alternative to this PR.
And [yajl](https://github.com/anjohnson/yajl) also supports JSON5 and stream parsing.

**Comment by jb-perrier** [comment]
As discussed on discord with @lerno , we should not add any dependency to C3. I've removed the whole json creation helpers funcs i was planning, to keep complexity and code size low. I guess when we will rewrite the compiler in C3 a complete JSON stdlib would be written.

File: src/utils/stringutils.c [hunk 1]
```diff
@@ -232,6 +232,27 @@ void scratch_buffer_append_signed_int(int64_t i)
 	scratch_buffer.len += len_needed;
 }
 
+void scratch_buffer_append_double(double d)
+{
+	uint32_t len_needed = (uint32_t)sprintf(&scratch_buffer.str[scratch_buffer.len], "%f", d);
```
Comment (lerno) [inline src/utils/stringutils.c:237]: I seem to recall that %f has a fixed number of decimals. Are 6 decimals (or whatever it is) sufficient?

Comment (jb-perrier) [reply src/utils/stringutils.c:237]: Nice catch, i didn't know about that, we should use "%.15g" instead.

**Pull request merged**